app/models/manageiq/providers/vmware/infra_manager/inventory/parser.rb

```diff
@@ -340,6 +355,22 @@ def parse_virtual_machine(object, kind, props)
     persister.vms_and_templates.targeted_scope << object._ref
     return if kind == "leave"
 
+    invalid, err = if props.fetch_path(:summary, :config).nil? || props.fetch_path(:config).nil?
+      [true, "Missing configuration for VM [#{object._ref}]"]
+    elsif props.fetch_path(:summary, :config, :uuid).blank? && props.fetch_path(:config, :uuid).blank?
+      [true, "Missing UUID for VM [#{object._ref}]"]
+    elsif props.fetch_path(:summary, :config, :vmPathName).blank?
+      [true, "Missing pathname location for VM [#{object._ref}]"]
+    else
+      false
+    end
+
+    if invalid
+      _log.warn("#{err} Skipping.")
+
```

Comment:
Super-ultra-minor, but there's a blank line here but not in the other one...probably should be consistent.